ui/cli-lib/src/Ariadne/UI/Cli/Face.hs

```diff
@@ -41,6 +45,11 @@ data UiLangFace = forall err expr. UiLangFace
 -- capture /what the UI can handle/, not what the backends can generate.
 data UiEvent
   = UiCommandEvent UiCommandId UiCommandEvent
+  | UiPasswordEvent UiPasswordEvent
+
+-- | Ui event triggered by the password manager
+data UiPasswordEvent
+  = UiPasswordRequest WalletReference CE.Event
```

Comment:
Type variable, I suppose. UI definitely shouldn't care, what kind of references does the backend use.

Reply:
If I use `data UiPasswordEvent walletRef = UiPasswordRequest walletRef CE.Event` then `UiEvent` will also need it. Are you sure with the type variable?